modules/objdetect/src/qrcode.cpp

```diff
@@ -863,43 +863,1423 @@ struct QRCodeDetector::Impl
+   for(size_t j = 0; j < hull.size(); j++)
+   {
+        Mat lineMask = Mat::zeros(bar.size(), threshold_output.type());
+        for(size_t i = 0; i < hull[j].size() - 1; i++)
+        {
+            line( lineMask, hull[j][i], hull[j][i + 1], 255, 1, 8, 0);
+        }
+        line( lineMask, hull[j][0], hull[j][hull[j].size() - 1], 255, 1, 8, 0);
+        vector<vector<Point> > mask_contours;
+        vector<Vec4i> mask_hierarchy;
+        findContours(lineMask, mask_contours, mask_hierarchy, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+        vector<Point2f> temp;
+        true_points_group.push_back(temp);
+        for(size_t i = 0; i < not_resized_loc_points.size(); i++)
+        {
+            double dist;
+            if((dist = pointPolygonTest(mask_contours[0], not_resized_loc_points[i], false)) > 0)
+            {
+              true_points_group[j].push_back(tmp_localization_points[i]);
+              tmp_localization_points[i].x = -1;
+            }
+        }
+    }
+    vector<Point2f> copy;
+    for(size_t j = 0; j < tmp_localization_points.size(); j++)
+    {
+       if(tmp_localization_points[j].x != -1) copy.push_back(tmp_localization_points[j]);
+    }
+
+    tmp_localization_points = copy;
+
+    for(int q = 0; q < num_qrcodes; q++)
+    {
+       for(size_t i = 0; i < true_points_group.size(); i++)
+       {
+          if (true_points_group[i].size() < 3)
+          {
+              for(size_t j = 0; j < true_points_group[i].size(); j++)
+                  tmp_localization_points.push_back(true_points_group[i][j]);
+              true_points_group[i].clear();
+          }
+       }
+     vector< vector< Point2f > > temp_for_copy;
+     for(size_t i = 0; i < true_points_group.size(); i++)
+     {
+          if(true_points_group[i].size() != 0)
+            temp_for_copy.push_back(true_points_group[i]);
+     }
+     true_points_group = temp_for_copy;
+     if(true_points_group.size() == 0)
+     {
+          true_points_group.push_back(tmp_localization_points);
+          tmp_localization_points.clear();
+     }
+     if (true_points_group[0].size() < 3) break;
+     num_points = 0;
+     for(size_t i = 0; i < true_points_group.size(); i++)
+     {
+         num_points = num_points + int(true_points_group[i].size());
+     }
+     num_points =  num_points + int(tmp_localization_points.size());
+     tmp_num_points = num_points;
+     int* set_size = new int[true_points_group.size()];
+     for(size_t i = 0; i < true_points_group.size(); i++)
+     {
+         set_size[i] = int(0.5 * (true_points_group[i].size() - 2 ) * (true_points_group[i].size() - 1));
+     }
+     vector< vector< vector< int> > > all_points(true_points_group.size());
+     for(size_t i = 0; i < true_points_group.size(); i++)
+         all_points[i].resize(set_size[i]);
+     for(size_t i = 0; i < true_points_group.size(); i++)
+        for (int j = 0; j < set_size[i]; j++)
+             all_points[i][j].resize(3);
+     int p = 0;
```

Comment:
what's mean variable `p`? please renamed variable, then other people understand it

Reply:
@allnes correct it